atomic_reactor/plugins/post_tag_and_push.py

```diff
@@ -95,11 +108,48 @@ def push_with_skopeo(self, registry_image, insecure, docker_push_secret):
             self.log.error("push failed with output:\n%s", e.output)
             raise
 
+    def source_get_unique_image(self):
+        source_result = self.workflow.prebuild_results[PLUGIN_FETCH_SOURCES_KEY]
+        koji_build_id = source_result['sources_for_koji_build_id']
+
+        kojisession = get_koji_session(self.workflow, NO_FALLBACK)
+        user_params = os.environ['USER_PARAMS']
+        user_data = json.loads(user_params)
+        container_target = user_data.get('koji_target')
+
+        timestamp = osbs.utils.utcnow().strftime('%Y%m%d%H%M%S')
+        random.seed()
+        current_platform = platform.processor() or 'x86_64'
+
+        tag_segments = [
+            container_target or 'none',
+            str(random.randrange(10**(RAND_DIGITS - 1), 10**RAND_DIGITS)),
+            timestamp,
+            current_platform
+        ]
+
+        tag = '-'.join(tag_segments)
+
+        get_build_meta = kojisession.getBuild(koji_build_id)
+        pull_specs = get_build_meta['extra']['image']['index']['pull']
+
+        source_image_spec = ImageName.parse(pull_specs[0])
```

Comment:
Do we need to `.enclose()` it first? Thinking about pre-Quay.io builds.

Reply:
added